### DISCUSSION TITLE
如何自动切换内外网？

### DISCUSSION THREAD

**Discussion by Godsing** [natfrp/wiki#560] [使用问题] [answered]
比如说，我现在新建了一条隧道：
xxx.natfrp.cloud:12345 -> local_ip:6666

我想要在手机 App 或电脑 web 访问 `xxx.natfrp.cloud:12345` 时，自动判断当前是否在局域网内：
- 如果不在局域网内，则使用 FRP 中转；
- 如果在局域网内，就自动直连 local_ip:6666 而不经过 FRP 中转。

有什么办法吗？

我想过在路由器添加一条 hosts 记录，但那样只能将 xxx.natfrp.cloud 映射到 local_ip，没法映射端口。

[a comment by an automated account is not shown]

**Comment by Christopher-Songs** [comment]
我也有类似的问题，不过更复杂一点。普通的http映射，我在openwrt上简单映射一下可以将域名+端口转发到目标设备，但对于https，暂时没想到方法。
我是https国内节点映射，并且开启了子域绑定，但本地服务本身是http的，想知道这种怎么玩？

[a comment by an automated account is not shown]
